src/spurt/workflows/emcf/_unwrap.py

```diff
@@ -0,0 +1,88 @@
+import json
+from pathlib import Path
+
+import h5py
+import numpy as np
+
+import spurt
+
+from ._settings import GeneralSettings, SolverSettings
+from ._solver import EMCFSolver
+
+logger = spurt.utils.logger
+
+
+def unwrap_tiles(
+    stack: spurt.io.SLCStackReader,
+    g_time: spurt.graph.PlanarGraphInterface,
+    gen_settings: GeneralSettings,
+    solv_settings: SolverSettings,
+) -> None:
+    """Unwrap each tile and save to h5."""
+    # Temporal graph
+    s_time = spurt.mcf.ORMCFSolver(g_time)  # type: ignore[abstract]
+
+    with gen_settings.tiles_jsonname.open(mode="r") as fid:
+        tiledata = json.load(fid)
+
+    # Iterate over tiles
+    for tt in range(len(tiledata["tiles"])):
+        tfname = str(gen_settings.tile_filename(tt))
+        if Path(tfname).is_file():
+            logger.info(f"Tile {tt+1} already processed. Skipping...")
+            continue
+
+        # Select valid pixels from coherence file
+        logger.info(f"Processing tile: {tt+1}")
+        space = _bounds_to_space(tiledata["tiles"][tt]["bounds"])
+        coh = stack.read_temporal_coherence(space)
+
+        # Create spatial graph and solver
+        g_space = spurt.graph.DelaunayGraph(
+            np.column_stack(np.nonzero(coh > stack.temp_coh_threshold))
+        )
+        s_space = spurt.mcf.ORMCFSolver(g_space)  # type: ignore[abstract]
+
+        # EMCF solver
+        solver = EMCFSolver(s_space, s_time, solv_settings)
+        wrap_data = stack.read_tile(space)
+        assert wrap_data.shape[1] == g_space.npoints
+        logger.info(f"Time steps: {solver.nifgs}")
+        logger.info(f"Number of points: {solver.npoints}")
+
+        uw_data = solver.unwrap_cube(wrap_data.data)
```

Comment:
This line errors for me when I try to run the command line workflow:

```
$ python -m spurt.workflows.emcf -i . -w 3
2024-05-21 17:24:19,342 [26507] [INFO] spurt: Using Hop3 Graph in time with 21 epochs.
2024-05-21 17:24:19,343 [26507] [INFO] spurt: Using existing tiles file: emcf_tmp/tiles.json
2024-05-21 17:24:19,343 [26507] [INFO] spurt: Processing tile: 1
2024-05-21 17:24:31,803 [26507] [INFO] spurt: Time steps: 57
2024-05-21 17:24:31,803 [26507] [INFO] spurt: Number of points: 421201
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/staniewi/repos/spurt/src/spurt/workflows/emcf/__main__.py", line 5, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/staniewi/repos/spurt/src/spurt/workflows/emcf/_cli.py", line 96, in main
    unwrap_tiles(stack, g_time, gen_settings, slv_settings)
  File "/Users/staniewi/repos/spurt/src/spurt/workflows/emcf/_unwrap.py", line 53, in unwrap_tiles
    uw_data = solver.unwrap_cube(wrap_data.data)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/staniewi/repos/spurt/src/spurt/workflows/emcf/_solver.py", line 115, in unwrap_cube
    if wrap_data.time_dim != 0:
       ^^^^^^^^^^^^^^^^^^
AttributeError: 'numpy.ndarray' object has no attribute 'time_dim'
```

It looks like `solver.unwrap_cube` wants and `Irreg3DInput` as input, but `wrap_data.data` is the numpy array contained in that class.

Reply:
Try changing that to just `wrap_data` instead of `wrap_data.data`. I plan to catch up with all the comments tomorrow. Since the test dataset was so big, I tested it stage by stage. There is a finite chance that I switched something after I went past that processing step

Reply:
That fixed it. This has been committed to `feat/dirichlet`